bokeh/models/annotations/html/labels.py

```diff
@@ -0,0 +1,329 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2021, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+'''
+
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Bokeh imports
+from ....core.enums import (
+    AngleUnits,
+    FontStyle,
+    SpatialUnits,
+    TextAlign,
+    VerticalAlign,
+)
+from ....core.properties import (
+    Alpha,
+    Angle,
+    AngleSpec,
+    Color,
+    Datetime,
+    Enum,
+    Float,
+    Include,
+    NonNullable,
+    NumberSpec,
+    Override,
+    String,
+    StringSpec,
+    field,
+)
+from ....core.property_mixins import (
+    FillProps,
+    LineProps,
+    ScalarFillProps,
+    ScalarLineProps,
+    ScalarTextProps,
+    TextProps,
+)
+from ....model import Qualified
+from ....util.serialization import convert_datetime_type
+from ..annotation import DataAnnotation
+from .html_annotation import HTMLAnnotation
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    "Label",
+    "LabelSet",
+    "Title",
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class Label(HTMLAnnotation, Qualified):
+    ''' Render a single HTML label as an annotation.
+
+    ``Label`` will render a single text label at given ``x`` and ``y``
+    coordinates, which can be in either screen (pixel) space, or data (axis
+    range) space.
+
+    The label can also be configured with a screen space offset from ``x`` and
+    ``y``, by using the ``x_offset`` and ``y_offset`` properties.
+
+    Additionally, the label can be rotated with the ``angle`` property.
+
+    There are also standard text, fill, and line properties to control the
+    appearance of the text, its background, as well as the rectangular bounding
+    box border.
+
+    See :ref:`userguide_annotations_labels` for information on plotting labels.
+
```

Comment:
I would request to repeat the admonition about HTML annotations and exports explicitly in every subclass (or maybe `HTMLAnnotation` could use ``__init_subclass__` to automate adding such?

Reply:
I will leave this for a documentation specific PR.